File: WebAPI/DecorVistaApi/DecorVistaApi/Services/DesignerService.cs
using DecorVistaApi.Dtos;

namespace DecorVistaApi.Services;

public interface DesignerService
{
    public bool Register(DesignerDto designerdto);
    public List<DesignerDto> FindAll();
    public DesignerDto FindById(int id);
    public DesignerDto FindByEmail(string email);
    public bool Login(DesignerDto designerdto);

}

File: Web/src/app/user/services/designer.service.ts
import { HttpClient } from "@angular/common/http";
import { Injectable } from "@angular/core";

import { lastValueFrom } from "rxjs";
import { Data } from "@angular/router";
import { BaseURLService } from "./baseURL.service";

@Injectable({
    providedIn: 'root',

})
export class DesignerService{
    constructor(
        private httpClient: HttpClient,
        private baseUrlService : BaseURLService
    ){}
    async register(formData: FormData){
        return lastValueFrom(this.httpClient.post(this.baseUrlService.BASE_URL + 'designer/register', formData));
    }
    async findbyemail(email: string){
        return lastValueFrom(this.httpClient.get(this.baseUrlService.BASE_URL + 'designer/findbyemail/'+email));
    }
    async login(email: string, password:string){
        return lastValueFrom(this.httpClient.post(this.baseUrlService.BASE_URL + 'designer/login/',{email,password}));
    }
}